File: Hashmap/Group Anagrams/GroupAnagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        Dictionary<string, List<string>> group = new Dictionary<string, List<string>>();
        
        foreach (string element in strs) {
            string sortedElement = new string(element.ToCharArray().OrderBy(c => c).ToArray());
            
            if (!group.ContainsKey(sortedElement)) {
                group[sortedElement] = new List<string> { element };
            } else {
                group[sortedElement].Add(element);
            }
        }
        
        IList<IList<string>> result = new List<IList<string>>();
        foreach (var kv in group) {
            result.Add(kv.Value);
        }
        
        return result;
    }
}

File: Hashmap/Group Anagrams/GroupAnagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> group;
        
        for (const string& element : strs) {
            string sortedElement = element;
            sort(sortedElement.begin(), sortedElement.end());
            if (group.find(sortedElement) == group.end()) {
                group[sortedElement] = {element};
            } else {
                group[sortedElement].push_back(element);
            }
        }

        vector<vector<string>> result;
        for (const auto& kv : group) {
            result.push_back(kv.second);
        }
        
        return result;
    }
};